src/main/java/net/neoforged/moddevgradle/internal/ModDevRunWorkflow.java

```diff
@@ -0,0 +1,490 @@
+                    if (modulePathDependency != null) {
+                        modulePath.getDependencies().add(modulePathDependency);
+                    }
+                },
+                legacyClassPath -> legacyClassPath.extendsFrom(additionalClasspath),
+                artifactsWorkflow.downloadAssets().flatMap(DownloadAssets::getAssetPropertiesFile),
+                project.provider(() -> versionCapabilities)
+        );
+    }
+
+    public static ModDevRunWorkflow get(Project project) {
+        var workflow = ExtensionUtils.findExtension(project, EXTENSION_NAME, ModDevRunWorkflow.class);
+        if (workflow == null) {
+            throw new InvalidUserCodeException("Please enable the modding plugin first by setting a version");
+        }
+        return workflow;
+    }
+
+    public static ModDevRunWorkflow create(Project project,
+                                           Branding branding,
+                                           ModDevArtifactsWorkflow artifactsWorkflow,
+                                           @Nullable ModuleDependency modulePathDependency,
+                                           @Nullable ModuleDependency runTypesConfigDependency,
+                                           @Nullable ModuleDependency testFixturesDependency,
+                                           ModuleDependency gameLibrariesDependency,
+                                           DomainObjectCollection<RunModel> runs,
+                                           VersionCapabilities versionCapabilites) {
+        var workflow = new ModDevRunWorkflow(
+                project,
+                branding,
+                artifactsWorkflow,
+                modulePathDependency,
+                runTypesConfigDependency,
+                testFixturesDependency,
+                gameLibrariesDependency,
+                runs,
+                versionCapabilites
+        );
+
+        project.getExtensions().add(EXTENSION_NAME, workflow);
+
+        return workflow;
+    }
+
+    public void configureTesting(Provider<ModModel> testedMod, Provider<Set<ModModel>> loadedMods) {
+        var testing = project.getExtensions().getByType(TestingExtension.class);
+        var testSuite = (JvmTestSuite) testing.getSuites().getByName("test");
+        var testSourceSet = testSuite.getSources();
+
+        var artifactsWorkflow = ModDevArtifactsWorkflow.get(project);
+        artifactsWorkflow.addToSourceSet(testSourceSet.getName());
+
+        var configurations = project.getConfigurations();
+
+        // If test fixtures are available for the current workflow, add them to runtime only
+        if (testFixturesDependency != null) {
+            configurations.getByName(testSourceSet.getRuntimeOnlyConfigurationName(), configuration -> {
+                configuration.getDependencies().add(testFixturesDependency);
+            });
+        }
+
+        for (var target : testSuite.getTargets()) {
+            setupTestTask(
```

Comment:
The current state of `setupTestTask` definitely does not support multiple test tasks and will fail miserably trying to configure more than one test task, we should change that or have a nicer error message.